### ISSUE TITLE
[1.12.2] (any) slot baubles causes baubes in necklase slot to unequip 

### ISSUE THREAD

**Issue by Shibva** [Corail31/tombstone_lite#60]
### Minecraft Version

1.12.2

### Forge Version

14.23.5.2860

### Corail Tombstone Version

tombstone-4.5.5-1.12.2.jar

### What happened?

I have 2 baubles, one that goes on the neck and one that can go in any slot 

when I retrieve my items the neclase I have one gets ejected from my full inventory while the bauble that can go in any slot seems to have gotten put into the location where the neclase was

I have the settings present and my guess is that the way it restores the inventory in those slots is using a simple process of just "shoving it back into the correct slots"

this works but it seems to cause ejection if you have one equipped bauble that can go into any slot

### Gametype

Singleplayer

### Happen by testing with only Corail Tombstone

No

### Other relevant mods

not really needed; just seems to be an issue regarding placement of items back into the bauble menu

### Valid Launcher

- [X] Yes

**Comment by Corail31** [comment]
@SteamDoge9000 You'll need to provide me a concrete example to make this happening with a minimum of mods (with the item names).
Tombstone uses the natural order to guess how to re-equip.

**Comment by Shibva** [comment]
oh, ok give me a bit to get you a minimum list of mod names to reproduce with

the pack I put together is large but if my thinking is correct what I will test real quick should prove to be beneficial for this list

**Comment by Shibva** [comment]
ok, tunrs out my thinking was right and was not with an mod that's in development 

it seems to work with any that's labeled as any; furthermore it seems todo the same if the item is soulbound

Botania is a mod that adds various fun baubles that have the (any) tag
so at the most this should be reproducible with only that mod 

all you need todo is to put one of them in any slot (minus the necalce slot, that would make it redudnantl. the Tainted blood pedant is a (amulet) so ensure you have one on you

then die and recover your stuff and you will find that it was ejected into the main inventory (or in the world if its full)

the soulbound enchantment does not appear to circumvent this on respawn either


it might not be the full minimum, but its apparently enough to reproduce with ease

let me know if this is still a problem with ya

**Comment by Shibva** [comment]
on a addional note it seems that any item that's soulound and a bauble will appear in the inventory first, if the inventory is full and you have baubles, some of your stuff may jetson out due to the baubles appearing in the main inventory first

not really a bug but something to point out and think about

**Comment by Corail31** [comment]
@SteamDoge9000 If it's specific to Botania (that no more support 1.12.2) and lots of items are equipable to any slot, there's nothing i will change on my side to retrieve the right slot because 1.12.2 is old and baubles slots don't have slot types for nothing.
NB: if it's specific to fewer items of another mod, i may have a look. As i'm not sure of the translation of your story, if you mean the baubles slots are empty and not equiped at all even though if you had baubles that could be equipped, then there's probably something i could look at. Can i sum up your story to "Botania items don't auto-equip in Baubles slots" ?

**Comment by Shibva** [comment]
well, that was just a example, the same could be done specifically if you had the item with a baubles tag of any. what im saying is that tehincaly speaking, any bauble supported item succmbs to this 

meaning if you were to make an item that was a babuble with any and another that had a neclase slot, you would find that it would happen with those items

the mod I found this bug in is in pre2 and can only be found on a discord.

in addition, finding mods that still support 1.12.2 still is a bit rare; more so with baubles so

if you want to test this fully, add dummy items or temporary attach bauble slot support to 2 of your items and test it

with that being said;  I just realized that this would also apply to EBwizardy, a mod that is still in 1.12.2 development and adds baubles (but none of them have the (any) tag, so the issue would not exist without it)

so the mods that you can use to test this is;
```Botania 
Electroblobs wizardry
blood magic
basically any mod that adds baubles
```

**Comment by Shibva** [comment]
basicaly the issue is with the any tag

it seems that how its done is that it assins it back into the correct slot. it works but with baubles with the (ANY) slot desiniton can go into any slot, meaning it would go into the first slot; the issue wth this is that any bauble that's equipped in that slot and the bauble with the (ANY) tag is on causes it it to overide the priority or whatever going into the first available slot, meaning any amulet that may have been equipt would become unequip as a result

TLDR: the system works but with how (ANY) works, it causes a unique issue 

possible ways to correct this may be extensive but the most simple that may run in my mind is for baubles that have (ANY) would be assigned to the slot that it was in to ensure that its put back into the slot it was last in

or if you want todo it simple, have baubles with (any) slot requirement be processed last

**Comment by Shibva** [comment]
Basicaly what im saying is this:

<img width="96" alt="image" src="https://user-images.githubusercontent.com/38770780/175664305-140a166f-2f3a-4f6b-a7d6-1cae084b3f61.png">
Lets say the green thing was an (Any) tag instead of (head) tag. if I was to die, the system would  put the bauble back into the slot they go into; but in the case of a bauble that can go into any slot (hence the ANY tag), the system would put the (ANY) bauble into the first slot it can go into rather than the slot it was last in. as a result the necklace item would be ejected out of the slot where the (ANY) tag bauble takes that slot instead, causing the necklace that was on to be unequipped and back into the inventory

hope the visual helped get the idea

**Comment by Shibva** [comment]
oh and to be clear; botania is NOT the only mod this happens to, I was just saying that it could be used as the only mod you would need to reproduce the issue with because it adds both baubles that are (amulets) and baubles that have the (ANY) tag

Edit: you could also do the same with what I suggested in adding temporary bauble slot abilities to  2 of your items (from what I've know, just giving it a tag would do the trick)

**Comment by Corail31** [comment]
I'm sorry but if you're not able to be concise, i can't help you as i don't understand your english.
Baubles type is this:
https://github.com/Azanor/Baubles/blob/master/src/main/java/baubles/api/BaubleType.java#L3
and the getter:
https://github.com/Azanor/Baubles/blob/master/src/main/java/baubles/api/IBauble.java#L19

**Comment by Shibva** [comment]
ok... let me start over, the second attempt

the bug/issue that is going on is that the item that I have in the baubles slots are not going into the right slot.

I found that the reason for this is bauble issue is baubles that can go into any of the slots (this is designated as {any) slot as you can see below)

I eventually realized from talking to you that this is happening because these types of baubles go into any slot. because of how I believe the mod handles items when recovering/respawning. the baubles that can go into any slot would instead go into the first slot; this happens to be the amulet slot. 

If I had an amulet equipped (that is referred to as Bauble A) and a bauble that can go into any slot provided as well (that is referred to as Bauble B), the mod would put bauble B into the Amulet slot; causing bauble A to not to reequipped correctly and go out into the inventory (or dropped as in it would do if your inventory is full)

Edit: a visual on what happens

Before death

After Death and recovering the items
(NOTE: the excitation point is a bauble that can go into any slot)

do you understand now what I'm trying to say; I hope I did better in explaining the situation now

**Comment by Shibva** [comment]
you can reopen it if you get the idea if ya want to 

if you need any more information let me know ^w^

**Comment by Corail31** [comment]
So, finally it's only trinkets being equiped before others slots, it'll be changed in dev but i can't say when is the next update for 1.12.2.
Edit: for soulbound, it's normal it's the same logic. And sorry, i was a bit slow to understand the relationship between the "any" and the "trinket" because i don't play except for tests ;)
I tested the new changes with some named items and all seem good for the next release, the change will also be added in 1.19 for Curios Api (and keeps the logic of natural order).

**Comment by Shibva** [comment]
thanks 
oh yeah, it also happens if the time is soulbound

thanks again

**Comment by Corail31** [comment]
@SteamDoge9000 You can now check if everything is ok in version 4.5.6.

**Comment by Shibva** [comment]
test seems to have done the trick, hopefully

**Comment by Shibva** [comment]
ok I foud out its doing it again, this time its just doing it in another slot; guess its not remembering the last slot it was in properly or something

**Comment by Shibva** [comment]
let me know if you see this message
@Corail31

**Comment by Shibva** [comment]
idk if your still getting notifications from this, but I hope you get this

**Comment by Corail31** [comment]
@Shibva i never say you to retrieve the exact slot, but to restore them based on the order they were dropped.
Your initial trouble was that an item wasn't equipped because another "trinket/any" was equipped before.
If it's again about an item being on the ground due to full inventory, in this case, explains the steps to make this happening.

**Comment by Shibva** [comment]
ok, I was going to say it's not happing now; but after a few creative tests, I found the issue is now only being caused by the soulbound enchantment system with any stot equipable baubles @Corail31. the issue that was reported initially is cleared up; however, it seems to be happening in regards to the soulbound enchantment now. Most notably that it will equip the bauble that's soulbound if you are not wearing it. Furthermore, it seems that the enchantment when replying is moving it into a different slot if it was equipped. as a result when opening a grave with bauble contents, a bauble that was previously equipped in that slot will fail as a result of it having one in that spot already being occupied.

this issue can be resolved by having soulbound baubles ensured that if they are equipped that they will be requipled in the last bauble slot it was in if the player were to die

Visuals demonstration

before death: (the shield-looking item is soulbound and can go into any bauble slot)
![image](https://user-images.githubusercontent.com/38770780/181120711-825644fb-96d4-4c2e-b6ab-c885e93f0833.png)

after death: (noticed the item has been moved when I resapwned)
![image](https://user-images.githubusercontent.com/38770780/181120793-a24259d6-60cf-41d3-9a2b-9be5cb55a5b1.png)

after recovering my grave
![image](https://user-images.githubusercontent.com/38770780/181120844-a68fae80-241d-43ca-9c9e-18f91b720786.png)

hope this visual helps get what im trying to say across. ^w^

Extra: false re-equiping (to just show that this could cause it as well)

![image](https://user-images.githubusercontent.com/38770780/181121169-5c90638f-1f0e-4f44-9908-a43dce3568db.png)

![image](https://user-images.githubusercontent.com/38770780/181121183-d4569394-6a9a-4f90-a067-e1b52764e103.png)


once this is fixed regarding soulbound, this issue and any re-equipping issues regarding it should be fixed in its entirety

feel free to reach out if you need more details ^w^

**Comment by Shibva** [comment]
Just to be clear, the previous issue is now cleared and its just soulbound now acting up;

here's proof of your implementation working accordingly with non-soulbound items
![image](https://user-images.githubusercontent.com/38770780/181121660-51ae2eb8-fa4e-45c2-889b-37a1128150bf.png)
![image](https://user-images.githubusercontent.com/38770780/181121668-ad850e7b-7c92-4b23-a101-2e7ef5f8c2e9.png)

Edit: while the one item moved, I can ensure that wont happen if an specified item is put into that slot; 

again, its only with any slot equipable baubles with soulbound causing an issue now

**Comment by Shibva** [comment]
I will make this a seperate issue to avoid further confustion
